core/src/main/scala/org/bitcoins/core/protocol/ln/LnParams.scala

```diff
@@ -37,11 +54,35 @@ object LnParams {
     override def lnRpcPort = 8080
 
     override def lnPort = 9735
+
+    override def invoicePrefix: ByteVector = {
+      ByteVector('l', 'n', 'b', 'c', 'r', 't')
+    }
   }
 
-  val allNetworks: Vector[LnParams] = Vector(LnBitcoinMainNet, LnBitcoinTestNet, LnBitcoinRegTest)
+  private val allNetworks: Vector[LnParams] = Vector(LnBitcoinMainNet, LnBitcoinTestNet, LnBitcoinRegTest)
 
   def fromNetworkParameters(np: NetworkParameters): Option[LnParams] = {
     allNetworks.find(_.network == np)
   }
+
+  private val prefixes: Map[String, LnParams] = {
+    val vec: Vector[(String, LnParams)] = {
+      allNetworks.map { network =>
+        (network.invoicePrefix.decodeAscii.right.get, network)
+      }
+    }
+    vec.toMap
+  }
+
+  /**
+   * Returns a [[org.bitcoins.core.protocol.ln.LnParams]] whose
+   * network prefix matches the given string. See BOLT11 for more details on prefixes
+   * [[https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md#human-readable-part]]
+   * @param str
+   * @return
+   */
+  def fromPrefixString(str: String): Option[LnParams] = {
+    prefixes.get(str)
```

Comment:
Can you help me understand why we have created the `prefixes` hash table, and are using it here? Is there additional functionality that you plan to use `prefixes` for, or could we replace all of this with  `allNetworks.find(_.invoicePrefix.decodeAscii.right.get == str)`?

Reply:
We could absolutely do that, but I get the feeling we might use that `Map` for other things, and we don't want to be copy and pasting the `allNetworks.find(_.invoicePrefix.decodeAscii.right.get == str)` as it is a pretty long nasty line of code. Better to have it in just one place.